File: cmd/gregale/templates/function-node24/handler.js
// Function template for gregale (node24).
//
// Mirrors cmd/gregale/templates/function-node/handler.js (node22) —
// the only differences are the runtime id (node24) and the handler
// filename (/app/node24.js inside the microVM, set by imaged's
// function-layer manifest). The CLI forces --runtime node24
// --handler handler.handler when deploying this template so the
// wiring is automatic; the `handler.handler` value in --handler is
// the customer's tarball stem (handler.js — the runner resolves the
// underlying filename per runtime).

export async function handler(event, ctx) {
  // Request events can contain cookies, API keys, authorization headers, and
  // customer payloads. Keep logs and the default response metadata-only.
  ctx.log.info("function invoked", { invocation_id: ctx.invocation_id, runtime: process.env.FAAS_RUNTIME });
  return {
    statusCode: 200,
    headers: { "content-type": "application/json" },
    body: JSON.stringify({
      ok: true,
      invocation_id: ctx.invocation_id,
      runtime: process.env.FAAS_RUNTIME,
    }),
  };
}

File: cmd/gregale/templates/function-python313/handler.py
"""Function template for gregale (python313).

Mirrors cmd/gregale/templates/function-python/handler.py (python312) —
the only difference is the runtime id (python313). The handler
filename is identical (/app/handler.py, version-neutral on the wire);
the underlying Python version is bound by the OCI base image
(images/runner-python313.Dockerfile), not by this file.

Functions are invoked directly by the runner — no HTTP server. The
CLI forces `--runtime python313 --handler handler.handler` when
deploying this template so the wiring is automatic.

`handler(event, ctx)` returns a dict with `statusCode`, optional
`headers`, and a string `body`. The runner ships the response back to
the gateway which forwards it to the customer.
"""
import json
import os


async def handler(event, ctx):
    # Do not log or echo event: it can contain cookies, API keys,
    # authorization headers, and customer payloads.
    ctx.log.info(
        "function invoked",
        extra={"invocation_id": ctx.invocation_id, "runtime": os.environ.get("FAAS_RUNTIME")},
    )
    return {
        "statusCode": 200,
        "headers": {"content-type": "application/json"},
        "body": json.dumps(
            {
                "ok": True,
                "invocation_id": ctx.invocation_id,
                "runtime": os.environ.get("FAAS_RUNTIME"),
            }
        ),
    }
